src/smt/smt_solver.cpp

```diff
@@ -61,16 +60,17 @@ void SmtSolver::finishInit(const LogicInfo& logicInfo)
     theory::TheoryConstructor::addTheory(d_theoryEngine.get(), id);
   }
   // Add the proof checkers for each theory
-  if (d_pnm)
+  ProofNodeManager* pnm = d_env.getProofNodeManager();
+  if (pnm)
   {
-    d_theoryEngine->initializeProofChecker(d_pnm->getChecker());
+    d_theoryEngine->initializeProofChecker(pnm->getChecker());
```

Comment:
Shouldn't we check `env.isTheoryProofProducing()` instead?

Reply:
No, this is actually a key realization we had: the theory engine contains ways of checking e.g. Boolean rules that need to be enabled for preprocessing even if the TheoryEngine isnt producing proofs.